Issue with char-server disconnecting and in-game making a party with ""

Hercules Git Version: 37fefe0d863f74a8229dd57c72a6cef6c1071730

When someone makes a party with /organize "NAME", with the "", and do something that need to check the party (like doing alt + z, or logout), it can not logout (character select menu) and the char-server is disconnected:
[Status]: Char-server 'SERVER' has disconnected.
[Status]: Attempting to connect to Char Server. Please wait.
[Status]: Connecting to 127.0.0.1:6121
[Error]: make_connection: connect failed (socket #6, error 111: Connection refused)!
[Status]: Connecting to 127.0.0.1:6121
[Error]: make_connection: connect failed (socket #6, error 111: Connection refused)!
[...]

Can you please tell me how to do that? I tryed to follow the "http://herc.ws/wiki/GDB" tutorial, but did not worked. Also, when i recompile the server, i noted that an error shows up a lot of time, like that:

[...]
      CC      nullpo.c
        CC      random.c
        CC      showmsg.c
        CC      strlib.c
strlib.c: In function 'strlib_normalize_name':
strlib.c:193: warning: logical '&&' with non-zero constant will always evaluate as true
strlib.c:203: warning: logical '&&' with non-zero constant will always evaluate as true
strlib.c:210: warning: logical '&&' with non-zero constant will always evaluate as true
strlib.c: In function 'strlib_e_mail_check':
strlib.c:314: warning: logical '&&' with non-zero constant will always evaluate as true
strlib.c: In function 'sv_escape_c':
strlib.c:767: warning: logical '&&' with non-zero constant will always evaluate as true
        CC      sysinfo.c
        CC      thread.c
        CC      timer.c
        CC      utils.c
[...]

Could it be related?

--- a/src/char/int_party.c
+++ b/src/char/int_party.c
@@ -485,7 +485,7 @@ int mapif_parse_CreateParty(int fd, const char *name, int item, int item2, const
 	int i;
 	nullpo_ret(name);
 	nullpo_ret(leader);
-	if( (p=inter_party->search_partyname(name))!=NULL){
+	if (!*name || (p = inter_party->search_partyname(name)) != NULL) {
 		mapif->party_created(fd,leader->account_id,leader->char_id,NULL);
 		return 0;
 	}
@@ -496,7 +496,8 @@ int mapif_parse_CreateParty(int fd, const char *name, int item, int item2, const
 				if (name[i] == '"') { /* client-special-char */
 					char *newname = aStrndup(name, NAME_LENGTH-1);
 					normalize_name(newname,"\"");
-					mapif->parse_CreateParty(fd,name,item,item2,leader);
+					trim(newname);
+					mapif->parse_CreateParty(fd, newname, item, item2, leader);
 					aFree(newname);
 					return 0;
 				}
@@ -513,7 +514,7 @@ int mapif_parse_CreateParty(int fd, const char *name, int item, int item2, const
 
 	p = (struct party_data*)aCalloc(1, sizeof(struct party_data));
 
-	memcpy(p->party.name,name,NAME_LENGTH);
+	safestrncpy(p->party.name, name, NAME_LENGTH);
 	p->party.exp=0;
 	p->party.item=(item?1:0)|(item2?2:0);
 

